components/shared/Mentor/Chat.tsx

```diff
@@ -13,59 +13,85 @@ import { ChatProps, threadType } from "@/types/mentor";
 import { transformChatData } from "@/lib/utils";
 import Image from "next/image";
 
-const Chat = ({ userMessages, toggleDrawer, userDetails }: ChatProps) => {
+const Chat = ({ userMessages, toggleDrawer, userDetails, setUserMessages }: ChatProps) => {
   // text is where the text box saves what the mentor writes
   const [text, setText] = useState<string>("");
 
   const chatData = transformChatData(userMessages?.messages);
 
-  console.log("transformed messages", chatData);
-
   // an empty div at the end of the thread used to scroll to the bottom on send
   const bottomOfPanelRef = useRef<HTMLDivElement | null>(null);
 
   // a referance for where the you will write the text
   const textBox = useRef<HTMLInputElement | null>(null);
 
-  // useEffect(() => {
-  //   // Scroll to the bottom whenever `thread` changes
-  //   if (bottomOfPanelRef.current) {
-  //     bottomOfPanelRef.current.scrollIntoView({ behavior: "smooth" }); // Optional: Add smooth scrolling
-  //   }
-  // }, [userMessages]);
-
-  // const sendText = (messageText: string) => {
-  //   // adds a message from the mentor to the thread and sets the text box to empty
-
-  //   setThread((prevThread) => [...prevThread, {
-  //     isUser: false,
-  //     text: messageText,
-  //     time: '2:44 PM',
-  //     newDay: '',
-  //     id: uuidv4()
-  //   }])
-
-  //   if (textBox.current) {
-  //     textBox.current.value = ''
-  //   }
-  // }
+  const sendText = async (messageText: string) => {
+    if (!messageText.trim()) return; // Don't send empty messages
+
+    const newMessage = {
+      sender: "mentor",
+      dateTime: new Date().toISOString(), // Get current time
+      content: messageText.trim(),
+    };
+
+    try {
+      // Find the current id from props
+      if (!userMessages) {
+        return;
+      }
+
+      // Append the new message to the messages array
+      const updatedMessages = [...userMessages.messages, newMessage];
+
+      // Update the backend
+      const patchResponse = await fetch(`http://localhost:3001/userMessages/${userMessages.id}`, {
+        method: "PATCH",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ messages: updatedMessages }),
+      });
+
+      console.log("the patch response", patchResponse)
```

Comment:
😒 how you handle error ?

Reply:
I was using a json server to mock the backend response but now I have been informed that the backend schma will change a lot from my implementation here and i will need to completely redo the fetching implementation. 
I will have to change the code and i will add error handling when i do that.